### PR TITLE
refactor ui framework

### PR DESCRIPTION
**Pull request by zilongshanren**

### CODE REVIEW

File: cocos/ui/UITextAtlas.cpp [hunk 1]
```diff
@@ -91,25 +91,33 @@ void TextAtlas::setProperty(const std::string& stringValue, const std::string& c
...
+//    CCLOG("cssss w %f, h %f", _contentSize.width, _contentSize.height);
 }
 
-const std::string& TextAtlas::getStringValue() const
+const std::string& TextAtlas::getString() const
 {
     return _labelAtlasRenderer->getString();
 }
+    
+ssize_t TextAtlas::getStringLength()
```
Comment (dumganhar) [inline cocos/ui/UITextAtlas.cpp]: `getter` should be a const function

Comment (zilongshanren) [inline cocos/ui/UITextAtlas.cpp]: fixed

File: cocos/ui/UILoadingBar.cpp [hunk 2]
```diff
@@ -120,15 +116,11 @@ void LoadingBar::setBarDirection(cocos2d::ui::LoadingBar::Direction direction)
 
 }
 
-LoadingBar::Direction LoadingBar::getBarDirection()
+LoadingBar::Direction LoadingBar::getDirection()
```
Comment (minggo) [inline cocos/ui/UILoadingBar.cpp]: LoadingBar::Direction LoadingBar::getDirection() -> LoadingBar::Direction LoadingBar::getDirection() const

File: cocos/ui/UITextAtlas.h [hunk 3]
```diff
@@ -73,10 +73,21 @@ class TextAtlas : public Widget
...
+    const std::string& getString() const;
+    
+    /**
+     * Gets the string length of the label.
+     * Note: This length will be larger than the raw string length,
+     * if you want to get the raw string length, you should call this->getString().size() instead
+     *
+     * @return  string length.
+     */
+    ssize_t getStringLength();
```
Comment (minggo) [inline cocos/ui/UITextAtlas.h]: ssize_t getStringLength() -> ssize_t getStringLength() const

File: tests/cpp-tests/Classes/UITest/CocoStudioGUITest/CocosGUIScene.cpp [hunk 4]
```diff
@@ -16,8 +16,22 @@ static struct
 }
 g_guisTests[] =
 {
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID || CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
+    {
+        "VideoPlayerTest",
+        [](Ref* sender)
+        {
+            UISceneManager* pManager = UISceneManager::sharedUISceneManager();
```
Comment (minggo) [inline tests/cpp-tests/Classes/UITest/CocoStudioGUITest/CocosGUIScene.cpp]: Don't use Hungary style.

File: cocos/ui/UILoadingBar.cpp (hunk 2 above)
Comment (zilongshanren) [inline cocos/ui/UILoadingBar.cpp]: ok

File: tests/cpp-tests/Classes/UITest/CocoStudioGUITest/CocosGUIScene.cpp (hunk 4 above)
Comment (zilongshanren) [inline tests/cpp-tests/Classes/UITest/CocoStudioGUITest/CocosGUIScene.cpp]: got it.

**Pull request merged**